ckanext/privatedatasets/actions.py

```diff
@@ -52,11 +52,97 @@ def package_acquired(context, request_data):
     :rtype: dict
 
     '''
+    context['method'] = 'grant'
+    return _process_package(context, request_data)
 
+def acquisitions_list(context, data_dict):
+    '''
+    API to retrieve the list of datasets that have been acquired by a certain user
+
+    :parameter user: The user whose acquired dataset you want to retrieve. This parameter
+        is optional. If you don't include this identifier, the system will use the one
+        of the user that is performing the request
+    :type user: string
+
+    :return: The list of datarequest that has been acquired by the specified user
+    :rtype: list
+    '''
+
+    if data_dict is None:
+        data_dict = {}
+
+    if 'user' not in data_dict and 'user' in context:
+        data_dict['user'] = context['user']
+
+    plugins.toolkit.check_access(constants.ACQUISITIONS_LIST, context.copy(), data_dict)
+
+    # Init db
+    db.init_db(context['model'])
+
+    # Init the result array
+    result = []
+
+    # Check that the user exists
+    try:
+        plugins.toolkit.get_validator('user_name_exists')(data_dict['user'], context.copy())
+    except Exception:
+        raise plugins.toolkit.ValidationError('User %s does not exist' % data_dict['user'])
+
+    # Get the datasets acquired by the user
+    query = db.AllowedUser.get(user_name=data_dict['user'])
+
+    # Get the datasets
+    for dataset in query:
+        try:
+            dataset_show_func = 'package_show'
+            func_data_dict = {'id': dataset.package_id}
+            internal_context = context.copy()
+
+            # Check that the the dataset can be accessed and get its data
+            # FIX: If the check_access function is not called, an exception is risen.
+            plugins.toolkit.check_access(dataset_show_func, internal_context, func_data_dict)
+            dataset_dict = plugins.toolkit.get_action(dataset_show_func)(internal_context, func_data_dict)
+
+            # Only packages with state == 'active' can be shown
+            if dataset_dict.get('state', None) == 'active':
+                result.append(dataset_dict)
+        except Exception:
+            pass
+
+    return result
+
+def package_deleted(context, request_data):
```

Comment:
Change the name to the function. The package is not deleted here but the user is removed from the list of allowed users.